client/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FC, createContext, useState, useMemo } from 'react';
+
+export type ThemeContextData = {
+  darkMode: boolean;
+  icon: boolean;
+  setDarkMode: (darkMode: boolean) => void;
+  setIcon: (icon: boolean) => void;
+};
+
+const DarkLightThemeContext = createContext<ThemeContextData>({
+  darkMode: false,
+  icon: false,
+  setDarkMode: () => {
+    return null;
+  },
+  setIcon: () => {
+    return null;
+  }
+});
+
+export const ThemeProvider: FC<React.ReactNode> = ({ children }) => {
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+  const [icon, setIcon] = useState<boolean>(false);
+
+  const darkLightThemeContext = useMemo(() => {
+    return { darkMode, setDarkMode, icon, setIcon };
+  }, [darkMode, icon]);
```

Comment:
no need to use memo here

Reply:
The object passed as the value prop to the Context provider changes every render. To fix this consider wrapping it in a useMemo hook.  eslint[react/jsx-no-constructed-context-values]

Reply:
ok, leave as is to not bother eslint :)